int8 direction0
int8 direction1
int8 distance0
int8 distance1
int8 ticks0
int8 ticks1